Difficulty running e2e tests

I keep running into an error running the end to end tests on the master branch with the following failure:
```
==> deploy initial
Running 'kapp deploy -f - -a test-template --diff-changes --tty -n kapp-test --into-ns kapp-test --yes'...
Running 'kapp delete -a test-template -n kapp-test --yes'...
--- FAIL: TestTemplate (610.06s)
    kapp.go:83: Failed to successfully execute 'kapp deploy -f - -a test-template --diff-changes --tty -n kapp-test --into-ns kapp-test --yes': Execution error: stdout: '--- create configmap/config-ver-1 (v1) namespace: kapp-test
              0 + apiVersion: v1
              1 + data:
              2 +   key1: val1
              3 + kind: ConfigMap
              4 + metadata:
              5 +   annotations:
              6 +     kapp.k14s.io/versioned: ""
              7 +   labels:
              8 +     kapp.k14s.io/app: "1574793934192273000"
              9 +     kapp.k14s.io/association: v1.62442fcb57d209abe192460ff4403c7c
             10 +   name: config-ver-1
             11 +   namespace: kapp-test
             12 +
        --- create secret/secret-ver-1 (v1) namespace: kapp-test
              0 + apiVersion: v1
              1 + data:
              2 +   key1: val1
              3 + kind: Secret
              4 + metadata:
              5 +   annotations:
              6 +     kapp.k14s.io/versioned: ""
              7 +   labels:
              8 +     kapp.k14s.io/app: "1574793934192273000"
              9 +     kapp.k14s.io/association: v1.8275953a67aea7cbb147343af1608f5b
             10 +   name: secret-ver-1
             11 +   namespace: kapp-test
             12 +
        --- create deployment/dep (apps/v1) namespace: kapp-test
              0 + apiVersion: apps/v1
              1 + kind: Deployment
              2 + metadata:
              3 +   labels:
              4 +     kapp.k14s.io/app: "1574793934192273000"
              5 +     kapp.k14s.io/association: v1.6f33d83389b69bd6ab35a82aa23e12fc
              6 +   name: dep
              7 +   namespace: kapp-test
              8 + spec:
              9 +   replicas: 1
             10 +   selector:
             11 +     matchLabels:
             12 +       app: dep
             13 +       kapp.k14s.io/app: "1574793934192273000"
             14 +   template:
             15 +     metadata:
             16 +       labels:
             17 +         app: dep
             18 +         kapp.k14s.io/app: "1574793934192273000"
             19 +         kapp.k14s.io/association: v1.6f33d83389b69bd6ab35a82aa23e12fc
             20 +     spec:
             21 +       containers:
             22 +       - args:
             23 +         - -listen=:80
             24 +         - -text=hello
             25 +         envFrom:
             26 +         - configMapRef:
             27 +             name: config-ver-1
             28 +         image: hashicorp/http-echo
             29 +         name: echo
             30 +         ports:
             31 +         - containerPort: 80
             32 +       volumes:
             33 +       - name: vol1
             34 +         secret:
             35 +           secretName: secret
             36 +

        Changes

        Namespace  Name          Kind        Conds.  Age  Op      Wait to    Rs  Ri
        kapp-test  config-ver-1  ConfigMap   -       -    create  reconcile  -   -
        ^          dep           Deployment  -       -    create  reconcile  -   -
        ^          secret-ver-1  Secret      -       -    create  reconcile  -   -

        Op:      3 create, 0 delete, 0 update, 0 noop
        Wait to: 3 reconcile, 0 delete, 0 noop

        10:45:35AM: ---- applying 3 changes [0/3 done] ----
        10:45:35AM: create configmap/config-ver-1 (v1) namespace: kapp-test
        10:45:35AM: create secret/secret-ver-1 (v1) namespace: kapp-test
        10:45:35AM: create deployment/dep (apps/v1) namespace: kapp-test
        10:45:36AM: ---- waiting on 3 changes [0/3 done] ----
        10:45:37AM: ok: reconcile configmap/config-ver-1 (v1) namespace: kapp-test
        10:45:37AM: ok: reconcile secret/secret-ver-1 (v1) namespace: kapp-test
        10:45:38AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:45:38AM:  ^ Waiting for 1 unavailable replicas
        10:45:38AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:45:38AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:45:38AM:     ^ Pending: ContainerCreating
        10:45:38AM: ---- waiting on 1 changes [2/3 done] ----
        10:46:38AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:46:38AM:  ^ Waiting for 1 unavailable replicas
        10:46:38AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:46:38AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:46:38AM:     ^ Pending: ContainerCreating
        10:46:39AM: ---- waiting on 1 changes [2/3 done] ----
        10:47:39AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:47:39AM:  ^ Waiting for 1 unavailable replicas
        10:47:39AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:47:39AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:47:39AM:     ^ Pending: ContainerCreating
        10:47:40AM: ---- waiting on 1 changes [2/3 done] ----
        10:48:39AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:48:39AM:  ^ Waiting for 1 unavailable replicas
        10:48:39AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:48:39AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:48:39AM:     ^ Pending: ContainerCreating
        10:48:40AM: ---- waiting on 1 changes [2/3 done] ----
        10:49:40AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:49:40AM:  ^ Waiting for 1 unavailable replicas
        10:49:40AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:49:40AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:49:40AM:     ^ Pending: ContainerCreating
        10:49:41AM: ---- waiting on 1 changes [2/3 done] ----
        10:50:40AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:50:40AM:  ^ Waiting for 1 unavailable replicas
        10:50:40AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:50:40AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:50:40AM:     ^ Pending: ContainerCreating
        10:50:41AM: ---- waiting on 1 changes [2/3 done] ----
        10:51:41AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:51:41AM:  ^ Waiting for 1 unavailable replicas
        10:51:41AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:51:41AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:51:41AM:     ^ Pending: ContainerCreating
        10:51:42AM: ---- waiting on 1 changes [2/3 done] ----
        10:52:42AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:52:42AM:  ^ Waiting for 1 unavailable replicas
        10:52:42AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:52:42AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:52:42AM:     ^ Pending: ContainerCreating
        10:52:43AM: ---- waiting on 1 changes [2/3 done] ----
        10:53:43AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:53:43AM:  ^ Waiting for 1 unavailable replicas
        10:53:43AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:53:43AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:53:43AM:     ^ Pending: ContainerCreating
        10:53:44AM: ---- waiting on 1 changes [2/3 done] ----
        10:54:44AM: ongoing: reconcile deployment/dep (apps/v1) namespace: kapp-test
        10:54:44AM:  ^ Waiting for 1 unavailable replicas
        10:54:44AM:  L ok: waiting on replicaset/dep-5658f778d6 (apps/v1) namespace: kapp-test
        10:54:44AM:  L ongoing: waiting on pod/dep-5658f778d6-mcksg (v1) namespace: kapp-test
        10:54:44AM:     ^ Pending: ContainerCreating
        10:54:45AM: ---- waiting on 1 changes [2/3 done] ----
        10:55:36AM: fail: reconcile deployment/dep (apps/v1) namespace: kapp-test

        ' stderr: 'Error: waiting on reconcile deployment/dep (apps/v1) namespace: kapp-test: finished unsuccessfully (Deployment is not progressing: ProgressDeadlineExceeded (message: ReplicaSet "dep-5658f778d6" has timed out progressing.))
        ' error: 'exit status 1'
```

When I describe the pod that is associated with this test I see the following:
```
Events:
  Type     Reason       Age                From                                                     Message
  ----     ------       ----               ----                                                     -------
  Normal   Scheduled    96s                default-scheduler                                        Successfully assigned kapp-test/dep-dc9bf645-bzrsp to gke-jims-playground-default-pool-477b1489-shzn
  Warning  FailedMount  32s (x8 over 96s)  kubelet, gke-jims-playground-default-pool-477b1489-shzn  MountVolume.SetUp failed for volume "vol1" : secrets "secret" not found
```

Is there some test setup that is necessary before running the end to end tests?

--- a/hack/test-all.sh
+++ b/hack/test-all.sh
@@ -3,6 +3,9 @@
 set -e -x -u
 
 ./hack/build.sh
+
+export KAPP_BINARY_PATH="$PWD/kapp"
+
 ./hack/test.sh
 ./hack/test-e2e.sh
 

--- a/hack/test-e2e.sh
+++ b/hack/test-e2e.sh
@@ -4,6 +4,8 @@ set -e -x -u
 
 go clean -testcache
 
+export KAPP_BINARY_PATH="${KAPP_BINARY_PATH:-$PWD/kapp}"
+
 go test ./test/e2e/ -timeout 60m -test.v $@
 
 echo E2E SUCCESS

--- a/test/e2e/config_test.go
+++ b/test/e2e/config_test.go
@@ -11,7 +11,7 @@ import (
 func TestConfig(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 	kubectl := Kubectl{t, env.Namespace, logger}
 
 	config := `

--- a/test/e2e/create_update_delete_test.go
+++ b/test/e2e/create_update_delete_test.go
@@ -11,7 +11,7 @@ import (
 func TestCreateUpdateDelete(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 	kubectl := Kubectl{t, env.Namespace, logger}
 
 	yaml1 := `

--- a/test/e2e/diff_test.go
+++ b/test/e2e/diff_test.go
@@ -11,7 +11,7 @@ import (
 func TestDiff(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	yaml1 := `
 ---

--- a/test/e2e/env.go
+++ b/test/e2e/env.go
@@ -7,12 +7,19 @@ import (
 )
 
 type Env struct {
-	Namespace string
+	Namespace      string
+	KappBinaryPath string
 }
 
 func BuildEnv(t *testing.T) Env {
+	kappPath := os.Getenv("KAPP_BINARY_PATH")
+	if kappPath == "" {
+		kappPath = "kapp"
+	}
+
 	env := Env{
-		Namespace: os.Getenv("KAPP_E2E_NAMESPACE"),
+		Namespace:      os.Getenv("KAPP_E2E_NAMESPACE"),
+		KappBinaryPath: kappPath,
 	}
 	env.Validate(t)
 	return env

--- a/test/e2e/ignore_failing_api_services_flag_test.go
+++ b/test/e2e/ignore_failing_api_services_flag_test.go
@@ -11,7 +11,7 @@ import (
 func TestIgnoreFailingAPIServicesFlag(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	yaml1 := `
 ---

--- a/test/e2e/inspect_test.go
+++ b/test/e2e/inspect_test.go
@@ -11,7 +11,7 @@ import (
 func TestInspect(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	yaml1 := `
 ---

--- a/test/e2e/kapp.go
+++ b/test/e2e/kapp.go
@@ -13,6 +13,7 @@ import (
 type Kapp struct {
 	t         *testing.T
 	namespace string
+	kappPath  string
 	l         Logger
 }
 
@@ -46,8 +47,7 @@ func (k Kapp) RunWithOpts(args []string, opts RunOpts) (string, error) {
 
 	k.l.Debugf("Running '%s'...\n", k.cmdDesc(args, opts))
 
-	cmdName := "kapp"
-	cmd := exec.Command(cmdName, args...)
+	cmd := exec.Command(k.kappPath, args...)
 	cmd.Stdin = opts.StdinReader
 
 	var stderr, stdout bytes.Buffer

--- a/test/e2e/order_test.go
+++ b/test/e2e/order_test.go
@@ -9,7 +9,7 @@ import (
 func TestOrder(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	name := "test-order"
 	cleanUp := func() {

--- a/test/e2e/template_test.go
+++ b/test/e2e/template_test.go
@@ -13,7 +13,7 @@ import (
 func TestTemplate(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 	kubectl := Kubectl{t, env.Namespace, logger}
 
 	depYAML := `---

--- a/test/e2e/update_replace_test.go
+++ b/test/e2e/update_replace_test.go
@@ -8,7 +8,7 @@ import (
 func TestUpdateFallbackOnReplace(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 	kubectl := Kubectl{t, env.Namespace, logger}
 
 	yaml1 := `
@@ -86,7 +86,7 @@ spec:
 func TestUpdateAlwaysReplace(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 	kubectl := Kubectl{t, env.Namespace, logger}
 
 	yaml1 := `

--- a/test/e2e/update_retry_on_conflict_test.go
+++ b/test/e2e/update_retry_on_conflict_test.go
@@ -12,7 +12,7 @@ import (
 func TestUpdateRetryOnConflict_WithoutConflict(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	yaml1 := `
 ---
@@ -106,7 +106,7 @@ spec:
 func TestUpdateRetryOnConflict_WithConflict(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	yaml1 := `
 ---
@@ -194,7 +194,7 @@ spec:
 func TestUpdateRetryOnConflict_WithConflictRebasedAway(t *testing.T) {
 	env := BuildEnv(t)
 	logger := Logger{}
-	kapp := Kapp{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
 
 	yaml1 := `
 ---

--- a/test/e2e/version_test.go
+++ b/test/e2e/version_test.go
@@ -7,7 +7,7 @@ import (
 
 func TestVersion(t *testing.T) {
 	env := BuildEnv(t)
-	kapp := Kapp{t, env.Namespace, Logger{}}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, Logger{}}
 
 	out, _ := kapp.RunWithOpts([]string{"version"}, RunOpts{NoNamespace: true})
 

